crates/ty_python_semantic/src/types/subclass_of.rs

```diff
@@ -73,6 +77,32 @@ impl<'db> SubclassOfType<'db> {
         !self.is_dynamic()
     }
 
+    pub(super) fn materialize(self, db: &'db dyn Db, variance: TypeVarVariance) -> Type<'db> {
+        match self.subclass_of {
+            SubclassOfInner::Dynamic(_) => match variance {
+                TypeVarVariance::Covariant => KnownClass::Type.to_instance(db),
+                TypeVarVariance::Contravariant => Type::Never,
+                TypeVarVariance::Invariant => {
+                    // We need to materialize this to `type[T]` but that isn't representable so
+                    // we instead use a type variable `T` with an upper bound of `type`.
+                    Type::TypeVar(TypeVarInstance::new(
+                        db,
+                        Name::new_static("T"),
```

Comment:
On second look, I think we should use a name other than `T` for these synthesized typevars; `T` is too commonly used in user code and too likely to cause confusion.

I think we could use the name `type[All]` here?

Reply:
I've used `T_all`